### PR TITLE
[feature] Optionally instrument generated travisCI with buildevents

### PR DESCRIPTION
**Pull request by edulop91**
### Summary
Instrument generated travisCI with [buildevents](https://github.com/honeycombio/buildevents)

### Test Plan
Apply in some of our internal repos

### References
- https://github.com/honeycombio/buildevents

### CODE REVIEW

File: templates/travis-ci/.travis.yml.tmpl [hunk 1]
```diff
@@ -27,18 +40,27 @@ jobs:
...
     - stage: check
       script:
         - set -e
+      {{ if $travisCI.Buildevents }}
+        - STEP_START=$(date +%s)
+        - STEP_SPAN_ID=$(echo check | sum | cut -f 1 -d \ )
+      {{ end }}
       {{- range $component := $testBucket }}
-        - pushd {{ $component.Dir }} && travis_retry make {{ $component.Command }} && popd
+        - {{ if $travisCI.Buildevents }} buildevents cmd $TRAVIS_BUILD_ID $STEP_SPAN_ID {{ $component.Dir }}.{{ $component.Command }} --{{ end }} pushd {{ $component.Dir }} && travis_retry make {{ $component.Command }} && popd
```
Comment (ryanking) [inline templates/travis-ci/.travis.yml.tmpl]: I wonder if we should try to capture a separate event for each try in the travis_retry's?

File: templates/travis-ci/.travis.yml.tmpl [hunk 2]
```diff
@@ -18,27 +39,55 @@ before_script:
...
             echo "git tree is dirty after running fogg apply"
             echo "ensure you run fogg apply on your branch"
             git status
             exit 1
           fi
-{{ range $testBucket := .TestBuckets }}
+          {{- if .Buildevents }}
+          buildevents step $BUILDEVENTS_BUILD_ID $STEP_SPAN_ID $STEP_START $STEP_NAME
+          {{- end }}
+{{- $travisCI := . }}
```
Comment (ryanking) [inline templates/travis-ci/.travis.yml.tmpl]: Is there a reason this is here and at the beginning of the file?

Comment (edulop91) [reply templates/travis-ci/.travis.yml.tmpl]: Good catch

File: templates/travis-ci/.travis.yml.tmpl [hunk 3]
```diff
@@ -1,15 +1,36 @@
...
+  - STEP_START=$(date +%s)
+  - STEP_NAME=install.${TRAVIS_JOB_NUMBER}
+  - STEP_SPAN_ID=$(echo ${STEP_NAME} | sum | cut -f 1 -d \ )
+  {{- end }}
   - pip install awscli --upgrade
   - aws --version
   - FOGG_VERSION="v{{ .FoggVersion }}" travis_retry make setup
+  {{- if .Buildevents }}
+  - mkdir bin # this is in theory supposed to exist already
+  - curl -L -o $HOME/bin/buildevents https://github.com/honeycombio/buildevents/releases/latest/download/buildevents-linux-amd64
```
Comment (ryanking) [inline templates/travis-ci/.travis.yml.tmpl]: Should we consider pinning the version? I know they have at least once broken the tool. Maybe just worry about that when it happens?

**Pull request merged**